packages/prepo-stores/src/utils/CustomProviders.ts

```diff
@@ -84,6 +84,10 @@ export class FallthroughProvider extends ethers.providers.BaseProvider {
       // alternatively, consider having an rpcFailed flag which triggers
       // a different UI when all RPCs are down (e.g. prompt user to add custom RPC)
       this.options.onError?.(e as Error)
+      if (e instanceof AggregateError) {
+        // eslint-disable-next-line no-console
+        e.errors.forEach((error) => console.warn(error))
+      }
```

Comment:
I think it should just throw the error object if it's not production, otherwise we might still miss an error warning from the console, wdyt?

```suggestion
      if (isProduction) {
        throw e
      } else {
        throw Error('No RPCs available at the moment.')
      }
```